### PR TITLE
improve logging and add flag

### PR DESCRIPTION
**Pull request by aireenmei**

### CODE REVIEW

File: src/maxdiffusion/configs/base_2_base.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ metrics_file: "" # for testing, local file that stores scalar metrics. If empty,
 write_metrics: True
 gcs_metrics: False
 metrics_period: 100  # calculate step time and loss
+metrics_buffer: false
```
Comment (suexu1025) [inline src/maxdiffusion/configs/base_2_base.yml]: @aireenmei can we resue write_metrics flag? if write_metrics set, we will do ping-pong metrics out. if not, no metrics file will be generated.

**Review by suexu1025 — COMMENTED** [review]
thanks, just a few comment.

File: src/maxdiffusion/configs/base_2_base.yml (hunk 1 above)
Comment (aireenmei) [reply src/maxdiffusion/configs/base_2_base.yml]: I currently use write_metrics=false to turn off any step time logging (including mllog) to get a clean baseline, that help access any logging overhead. See my comments https://buganizer.corp.google.com/issues/343225248#comment6

Comment (suexu1025) [reply src/maxdiffusion/configs/base_2_base.yml]: we already got two flags for metrics write out, `write_metrics`, `gcs_metrics`, we can apply buffer metrics as default for metrics write out. might not need to introduce new flag

File: src/maxdiffusion/models/train.py [hunk 2]
```diff
@@ -491,32 +491,43 @@ def compute_loss(unet_params):
...
-            _buffered_sample_count = total_train_batch_size * _buffered_step_num
+            step_num_delta = step_num - last_log_step_num
+
             # record metrics of current period
             record_scalar_metrics(train_metric, step_time_delta, step_num_delta, per_device_tflops, learning_rate_scheduler(step))
-            # print metrics of previous period
-            mllog_utils.maybe_train_step_log(config, start_step, _buffered_step_num, _buffered_sample_count, _buffered_metrics)
-            write_metrics(writer, local_metrics_file, running_gcs_metrics, train_metric, step, config)
+
+            if config.metrics_buffer:
```
Comment (suexu1025) [inline src/maxdiffusion/models/train.py]: while mllog for this branch mlperf  branch will always turned on. we can use write_metric here to make write metrics out as optional.

Comment (aireenmei) [reply src/maxdiffusion/models/train.py]: Now leaving only one flag, write_metrics for using buffer. The default behavior is no buffer with mllog on

File: src/maxdiffusion/configs/base_2_base.yml (hunk 1 above)
Comment (aireenmei) [reply src/maxdiffusion/configs/base_2_base.yml]: done

**Review by suexu1025 — APPROVED** [review]
Thanks, it is really helpful!

**Comment by aireenmei** [comment]
This run is with the updates: https://cloudlogging.app.goo.gl/MHkDvvhTpFrjmvjy8

**Pull request merged**